src/openloco/map/tilemgr.cpp

```diff
@@ -0,0 +1,18 @@
+#include "../interop/interop.hpp"
+#include "tilemgr.h"
+
+using namespace openloco::interop;
+
+namespace openloco::map::tilemgr
+{
+    static loco_global_array<tile_element*, 0x30004, 0x00E40134> _tiles;
+
+    tile get(coord_t x, coord_t y)
+    {
+        tile_coord_t tileX = x / 32;
+        tile_coord_t tileY = y / 32;
+
+        size_t index = ((y << 9) | x) >> 5;
```

Comment:
Don't you need to clear the low bits of y?

Reply:
Or just use tileX tileY

Reply:
This currently mimics the assembly which does not ensure out of bounds numbers. I can change it but I need to work out the map dimensions.